src/librbd/cache/ReplicatedWriteLog.cc

```diff
@@ -409,7 +607,50 @@ void ReplicatedWriteLog<I>::rwl_init(Context *on_finish, DeferredContexts &later
     } TX_FINALLY {
     } TX_END;
   } else {
-    // TODO: load existed cache. This will be covered in later PR.
+    m_cache_state->present = true;
+    /* Open existing pool */
+    if ((m_log_pool =
+         pmemobj_open(m_log_pool_name.c_str(),
+                      m_rwl_pool_layout_name)) == NULL) {
+      lderr(cct) << "failed to open pool (" << m_log_pool_name << "): "
+                 << pmemobj_errormsg() << dendl;
+      on_finish->complete(-errno);
+      return;
+    }
+    pool_root = POBJ_ROOT(m_log_pool, struct WriteLogPoolRoot);
+    if (D_RO(pool_root)->header.layout_version != RWL_POOL_VERSION) {
```

Comment:
@dillaman For now, we doesn't handle upgrading case. We'll handle upgrade RWL in the future. If you had an existing RWL pool and you restart w/ an updated librbd that bumped this version, you don't need to rollback. Our plan is that when a updated librbd that bumped this version, we'll use the data on the existing cache to construct the new version's log entry struct in the RAM, then drop the existing cache and create a new cache with the new version's log entry struct.

Reply:
That is: if the cache is in an old version, it will load the old data to build the data structures like m_log_entries, dirty_log_entries, etc. And trigger flush to flush all the cache data to OSD. After that reinitialize the cache file as empty.